File: 1161. Maximum Level Sum of a Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    unordered_map<int, int> s;
    
    void dfs(TreeNode* x, int dp) {
        if (!x) return;
        s[dp] += x->val;
        dfs(x->left, dp+1);
        dfs(x->right, dp+1);
    }
    
    int maxLevelSum(TreeNode* root) {
        dfs(root, 1);
        int dp, sum = INT_MIN;
        for (auto& kv : s)
            if (kv.second > sum || kv.second==sum && kv.first<dp) {
                dp = kv.first;
                sum = kv.second;
            }
        return dp;
    }
};

File: 1161. Maximum Level Sum of a Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def maxLevelSum(self, root: TreeNode) -> int:
        d = dict()
        def dfs(x : TreeNode, dp : int):
            if not x: return
            d[dp] = d.get(dp, 0) + x.val
            dfs(x.left, dp+1)
            dfs(x.right, dp+1)
        dfs(root, 1)
        ans, tot = 0, -sys.maxsize
        for dp, s in d.items():
            if s>tot or s==tot and dp<ans:
                ans, tot = dp, s
        return ans
            